File: week2/esadmsk_w2/evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        stack = []

        for token in tokens:
            if token.isdigit() or (token[0] == '-' and token[1:].isdigit()):
                stack.append(int(token))
            else:
                num2 = stack.pop()
                num1 = stack.pop()
                if token == '+':
                    stack.append(num1 + num2)
                elif token == '-':
                    stack.append(num1 - num2)
                elif token == '*':
                    stack.append(num1 * num2)
                elif token == '/':
                    stack.append(int(float(num1) / num2))

        return stack[-1]

File: week2/omerSefaYilmaz/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */

var evalRPN = function (tokens) {
    let stack = [];
  
    for (let i = 0; i < tokens.length; i++) {
      if (
        tokens[i] === "+" ||
        tokens[i] === "-" ||
        tokens[i] === "*" ||
        tokens[i] === "/"
      ) {
        const firstEl = stack.pop();
        const secondEl = stack.pop();
        const result = calculate(secondEl,firstEl, tokens[i]);
        stack.push(result);
      } else {
        stack.push(Number(tokens[i]));
      }
    }
    return stack.pop();
  };
  
  var calculate = (num1, num2, token) => {
    if (token === "+") return Number(num1) + Number(num2);
    if (token === "-") return Number(num1) - Number(num2);
    if (token === "*") return Number(num1) * Number(num2);
    if (token === "/") return num1 / num2 >= 0 ? Math.floor(num1 / num2) : Math.ceil(num1 / num2);
  };